### ISSUE TITLE
TensorRT return Error Code 1: Cuda Runtime (an illegal memory access was encountered) when mark intermediate tensor as output

### ISSUE THREAD

**Issue by FdyCN** [NVIDIA/TensorRT#3256]
## Description

I try to dump intermediate tensor by `mark_output` in ONNX-TensorRT model conversion. But when I used the converted TensorRT model and access the network outputs (with intermediate Tensor)，original output tensor is OK，but the marked intermediate Tensor access is WRONG with these log:

```
[TRT] Error: 1: [graphContext.h::nvinfer1::rt::MyelinGraphContext::~MyelinGraphContext::55] Error Code 1: Myelin (Error 700 destroying stream '000002993CB68C40'.)
 [TRT] Error: 1: [defaultAllocator.cpp::nvinfer1::internal::DefaultAllocator::deallocate::61] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [defaultAllocator.cpp::nvinfer1::internal::DefaultAllocator::deallocate::61] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
[TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaStream::~ScopedCudaStream::47] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
[TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
[TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
 [TRT] Error: 1: [cudaResources.cpp::nvinfer1::ScopedCudaEvent::~ScopedCudaEvent::24] Error Code 1: Cuda Runtime (an illegal memory access was encountered)
```

my `mark_output` codes in ONNX-TensorRT:

insert pisition:
 https://github.com/onnx/onnx-tensorrt/blob/0462dc31ae78f48744b6141ae376df1f96d3f459/ModelImporter.cpp#L628

codes:
```
    for (int i = 0; i < graph.node_size(); i ++){
        ::ONNX_NAMESPACE::NodeProto const& node = graph.node(i);

        if( node.output().size() > 0 && node.op_type() == "MatMul") {
            nvinfer1::ITensor* new_output_tensor_ptr = &_importer_ctx.tensors().at(node.output(0)).tensor();
            new_output_tensor_ptr->setName(node.output(0).c_str());
            _importer_ctx.network()->markOutput(*new_output_tensor_ptr);
        }
    }
```

Are these codes wrong or not in position？ 
But I can get these inserted intermediate Tensor during runtime by using this API： `nvinfer1::ICudaEngine::getIOTensorName(int32_t)`
## Environment

**TensorRT Version**: 8.5.1.7

**NVIDIA GPU**: RTX 4070 laptop

**NVIDIA Driver Version**: 536.25

**CUDA Version**: 11.8

**CUDNN Version**:8.9.1

Operating System:Windows 11

Python Version (if applicable):

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files

**Model link**:
https://huggingface.co/microsoft/beit-base-patch16-224-pt22k-ft22k

## Steps To Reproduce
reference: https://forums.developer.nvidia.com/t/tensorrt-encountered-issues-when-converting-weights-between-types-and-that-could-affect-accuracy/263946/2

It's my issue too. 

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: tried 8.6.1.6

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`): Not test

**Comment by FdyCN** [comment]
@ttyio Could you please take a look at this issue? Thank a lot.

**Comment by FdyCN** [comment]
forget about it. I found I did not allocate buffer with 256 bytes alignment....
